external/model-preparation-algorithm/mpa_tasks/utils/data_utils.py

```diff
@@ -44,34 +44,48 @@ def is_training_subset(subset):
         return subset.name in ["TRAINING", "VALIDATION"]
 
     def load_image_from_cache(filename: str, to_float32=False):
-        with open(filename, "rb") as f:
+        try:
+            f = open(filename, "rb")
             fcntl.flock(f, fcntl.LOCK_SH)
-            try:
-                cached_img = cv2.imread(filename)
-                if to_float32:
-                    cached_img = cached_img.astype(np.float32)
-                return cached_img
-            except Exception as e:
-                logger.warning(f"Skip loading cached {filename} \nError msg: {e}")
-            finally:
+            cached_img = np.asarray(bytearray(f.read()))
+            fcntl.flock(f, fcntl.LOCK_UN)
+            f.close()
+            cached_img = cv2.imdecode(cached_img, cv2.IMREAD_COLOR)
+            if to_float32:
+                cached_img = cached_img.astype(np.float32)
+            return cached_img
+        except Exception as e:
+            logger.warning(f"Skip loading cached {filename} \nError msg: {e}")
+            try:  # to handle the case that f is not defined: failed open(filename, "rb")
                 fcntl.flock(f, fcntl.LOCK_UN)
-    
+                f.close()
+            except Exception:
+                pass
```

Comment:
Can't we use `finally`?
Anyway, at least we can return `None` explicitly.

Reply:
Do you mean use `finally` instead of stacked `try~except`?
I intended to handle `f = open(filename, "rb")`, `f` will not be defined if failed to `open`.

Updated to return None explicitly! Thank you